### PR TITLE
Refactor code to follow flake8 style guide

### LINKED ISSUE #359: Add flake8 to pre-commit hooks
## Description

To have a consistent coding style, `flake8` coding style should be enforced. This ultimately includes adding it to the pre-commit hooks and CI.

### PR DESCRIPTION
**Pull request by WeilerP**
## New

* Global configuration file `.flake8`

## Changes

* `*.py` files are refactored and updated to follow `flake8` code style.
* `flake8` check is added to CI.
* Add `flake8` to pre-commit hooks.

## Related issues

Closes #359.

### CODE REVIEW

**Comment by WeilerP** [comment]
@VolkerBergen , `get_connectivities` is used [here](https://github.com/WeilerP/scvelo/blob/e7dc36e7e11a5848f2292ac8131f4cd7811792a9/scvelo/plotting/scatter.py#L536) but never imported. Should it be imported from `scv.pp.neighbors` or this part of the code dropped?

**Comment by VolkerBergen** [comment]
`get_connectivities` should be imported from `.utils`, as previously defined via `from .utils import *`.

**Comment by WeilerP** [comment]
> `get_connectivities` should be imported from `.utils`, as previously defined via `from .utils import *`.

Ah sorry, I had removed the import in `scvelo.plotting.utils` as it had not been used there and then forgot about it. Will add the import ASAP.

File: scvelo/plotting/utils.py [hunk 1]
```diff
@@ -907,8 +913,12 @@ def rgb_custom_colormap(colors=None, alpha=None, N=256):
 
     for j in range(ints):
         for i in range(3):
-            vals[n * j : n * (j + 1), i] = np.linspace(c[j][i], c[j + 1][i], n)
-        vals[n * j : n * (j + 1), -1] = np.linspace(alpha[j], alpha[j + 1], n)
+            start = n * j
```
Comment (VolkerBergen) [inline scvelo/plotting/utils.py:916]: `start` and `finish` can be defined before the second loop once instead.

File: scvelo/plotting/scatter.py [hunk 2]
```diff
@@ -1,10 +1,58 @@
...
+from matplotlib.colors import is_color_like
 import numpy as np
 import pandas as pd
+from pandas import unique
+
+from anndata import AnnData
+from scvelo import settings
+import scvelo.logging as logg
+from scvelo.preprocessing.neighbors import get_connectivities
+from .utils import (
```
Comment (VolkerBergen) [inline scvelo/plotting/scatter.py:14]: would prefer using `__all__` in utils and star import instead. What do you think?

File: scvelo/plotting/utils.py [hunk 3]
```diff
@@ -1465,15 +1476,14 @@ def log_fmt(x, pos):
...
-
+        for i, pdf_name in enumerate(pdf):
             xt = ax.get_xticks()
             xmin, xmax = min(xt), max(xt)
             lnspc = np.linspace(xmin, xmax, len(bins))
 
-            if "(" in pd:  # used passed parameters
-                args, pd = eval(pd[pd.rfind("(") :]), pd[: pd.rfind("(")]
+            if "(" in pdf_name:  # used passed parameters
+                start = pd.rfind("(")
```
Comment (VolkerBergen) [inline scvelo/plotting/utils.py:1485]: Also change to `pdf_name.rfind` (this is not a pandas function, but applied to a string).

File: scvelo/plotting/utils.py [hunk 4]
```diff
@@ -1465,15 +1476,14 @@ def log_fmt(x, pos):
...
+                args, pdf_name = eval(pd[start:]), pd[: pdf_name.rfind("(")]
```
Comment (VolkerBergen) [inline scvelo/plotting/utils.py:1486]: same here - should be `pdf_name`, right?

File: scvelo/tools/paga.py [hunk 5]
```diff
@@ -55,7 +54,9 @@ def get_sparse_from_igraph(graph, weight_attr=None):
 def set_row_csr(csr, rows, value=0):
     """Set all nonzero elements to the given value. Useful to set to 0 mostly."""
     for row in rows:
-        csr.data[csr.indptr[row] : csr.indptr[row + 1]] = value
+        start = csr.indptr[row]
+        finish = csr.indptr[row + 1]
```
Comment (VolkerBergen) [inline scvelo/tools/paga.py:58]: call it `end` instead?

File: tests/test_basic.py [hunk 6]
```diff
@@ -58,7 +58,7 @@ def test_pipeline():
 
     Ms, Mu = adata.layers["Ms"][0], adata.layers["Mu"][0]
     Vs, Vd = adata.layers["velocity"][0], adata.layers["dynamical_velocity"][0]
-    Vpca, Vgraph = adata.obsm["velocity_pca"][0], adata.uns["velocity_graph"].data[:5]
+    _, Vgraph = adata.obsm["velocity_pca"][0], adata.uns["velocity_graph"].data[:5]
```
Comment (VolkerBergen) [inline tests/test_basic.py:61]: just remove `adata.obsm["velocity_pca"][0]` (I think I originally wanted to have another assertion here).

File: scvelo/plotting/utils.py (hunk 3 above)
Comment (WeilerP) [reply scvelo/plotting/utils.py]: Yes, thanks. Good catch!

File: scvelo/tools/utils.py [hunk 7]
```diff
@@ -375,7 +375,8 @@ def test_bimodality(x, bins=30, kde=True, plot=False):
     )
 
     idx = int(bins / 2) - 2
-    idx += np.argmin(kde_grid[idx : idx + 4])
+    finish = idx + 4
```
Comment (VolkerBergen) [inline scvelo/tools/utils.py:378]: one more `finish` to be replaced with `end` ;)

**Pull request merged**